proptest/src/test_runner/runner.rs

```diff
@@ -178,38 +178,46 @@ fn call_test<V, F, R>(
     _runner: &mut TestRunner,
     case: V,
     test: &F,
-    replay: &mut R,
+    replay_from_fork: &mut R,
     result_cache: &mut dyn ResultCache,
     _: &mut ForkOutput,
-) -> TestCaseResult
+    is_from_persisted_seed: bool,
+) -> TestCaseResultV2
 where
     V: fmt::Debug,
     F: Fn(V) -> TestCaseResult,
     R: Iterator<Item = TestCaseResult>,
 {
-    if let Some(result) = replay.next() {
-        return result;
+    if let Some(result) = replay_from_fork.next() {
+        return result.map(|_| TestCaseOk::ReplaySuccess);
     }
 
     let cache_key = result_cache.key(&ResultCacheKey::new(&case));
     if let Some(result) = result_cache.get(cache_key) {
-        return result.clone();
+        return result.clone().map(|_| TestCaseOk::CacheHitSuccess);
     }
 
     let result = test(case);
     result_cache.put(cache_key, &result);
-    result
+    result.map(|_| {
+        if is_from_persisted_seed {
+            TestCaseOk::ReplaySuccess
```

Comment:
I don't think this is the right return value. we're actually running the test for this result so its not really a "replay" in the context of what "replay" means for forked tests. I commented on the enum def as well but to be explicit, what do you think about a `PersistedCaseSuccess` variant?

Reply:
good call out, will make that change 👍